frontend/ui/widget/dictquicklookup.lua

```diff
@@ -478,10 +433,16 @@ function DictQuickLookup:update()
                         else
                             self.ui:handleEvent(Event:new("Unhighlight"))
                         end
-                        self:update()
+                        -- Just update, repaint and refresh *this* button
+                        local this = self.button_table:getButtonById("highlight")
+                        if not this then return end
+                        this:enableDisable(self.highlight ~= nil)
+                        this:setText(self:getHighlightText(), this.width)
+                        this:refresh()
```

Comment:
I think there's an actual Highlight or something event sent right above that, and this one does repaint ReaderUI.

(I certainly see a ReaderUI repaint, so *something* is triggering it ^^).

Reply:
There's definitely a repaint with the Highlight/Unhighlight events.

There's no *refresh* though, which I would tend to agree with (e.g., what you originally selected to lookup is still visibly highlighted until you dismiss the popup).

Reply:
I just checked on master and there's actually no highighting change when hitting Highlight or Unhighlight - until the window is closed.
So forget my comment :) and keep the current behaviour (we can know how it will be from the button title).